templates/angular-class.mustache

```diff
@@ -11,13 +11,17 @@ angular.module('{{&moduleName}}', [])
      * @param {string} cache - An angularjs cache implementation
      */
     var {{&className}} = (function(){
-        function {{&className}}(domain, cache){
+        function {{&className}}(options, cache){
+            var domain = (typeof options === 'object') ? options.domain : options;
             this.domain = typeof(domain) === 'string' ? domain : '{{&domain}}';
             if(this.domain.length === 0) {
                 throw new Error('Domain parameter must be specified as a string.');
             }
-            this.domain = domain;
+            cache = cache || ((typeof options === 'object') ? options.cache : cache);
             this.cache = cache;
+            {{#isSecure}}
+                this.token = (typeof options === 'object') ? options.token : {};
```

Comment:
Ooppss :)
If the property exists it has to be {} because the setToken method try to access an object.